modules/openid_auth/app/controllers/openid_auth/v0/validation_controller.rb

```diff
@@ -15,8 +15,13 @@ def index
 
       private
 
+      def payload_object
+        @payload_object ||= OpenStruct.new(token_payload.merge(va_identifiers: { icn: nil }))
+      end
+
       def validated_payload
-        @validated_payload ||= OpenStruct.new(token_payload.merge(va_identifiers: { icn: @current_user.icn }))
+        payload_object.va_identifiers[:icn] = payload_object.try(:icn) || @current_user.icn
+        payload_object
```

Comment:
Seeing the memoization of `payload_object` here feels a little strange since `validated_payload` is the only caller and it modifies the value. My mind immediately wonders why we can't memoize `validated_payload` instead. I tried to recombine it and document how I read it. Unless my comments are wrong, it seems like keeping the logic together might be clearer.

```
def validated_payload
  # Memoize to avoid duplicate calls of the potentially-expensive User#mvi method.
  @validated_payload ||=
    begin
      # Ensure the token has an `va_identifiers` key.
      payload_object = OpenStruct.new(token_payload.merge(va_identifiers: { icn: nil }))

      # Sometimes we'll already have an `icn` in the token. If we do, copy if down into `va_identifiers`
      # for consistency. Otherwise use the ICN value we used to look up the MVI attributes.
      payload_object.va_identifiers[:icn] = payload_object.try(:icn) || @current_user.icn

      payload_object
    end

Reply:
Do we even need to memoize anything here?

Reply:
It doesn't seem to be saving us anything with the current implementation. The code I wrote out above was just a way to keep all of the elements of the original code. I figured _if_ we wanted to memoize anything it would be the entire operation, and especially the potential call to MVI. I'm cool with removing the memoization.

Reply:
Went ahead and removed it.